enhancements/etcd/etcd-tuning-profiles.md

```diff
@@ -0,0 +1,175 @@
+    - LEADER_ELECTION_TIMEOUT: 2500ms
+* Remove the parameters from the podspec rendering and replace with the Profile.
+* Add a config map to allow an adminstrator to change the profile.
+
+### Non-Goals
+
+* Adding more profiles beyond those listed above.
+* Add an API endpoint to change the profile.
+* Handle changing profile without a workload disruption.
+* Allow users to set arbitrary values for the profile parameters.
+
+## Proposal
+
+The profiles are a layer of abstraction that allow a customer to tweak etcd to run more reliably on their system, while not being so open as to allow them to easily harm their cluster by (knowingly or not) setting bad values for them.
+The values for the two proposed profiles for the Tech Preview of this feature are the current default values (applied to all platform except for) and the values applied for Azure and IBMCloud VPC.
+The latter values have been used successfully in the field for some time so the risk to future cluster is minimal.
+Changing to a "slower" profile will likely incur a performance/latency penalty, but that is likely an acceptable trade for cluster stability.
+
+In this iteration, for the Tech Preview, we will make it clear that changing the profile will require an etcd redeployment and thus a disruption on workloads. In the future enhancement, we can discuss a more seamless transition between profiles.
+We will not allow the user to set arbitrary values for the parameters, they must conform to the profiles values (by way of the profile).
+
+In the Tech Preview, the active profile will be set via a configmap that's applied before the etcd pod starts up. An administrator can update the profile by changing the configmap value and forcing the etcd pods to restart.
+
+### Workflow Description
+
+1. The cluster administrator decides to change the etcd profile from default to slow, or slow to default.
+2. They change the configmap with the new profile they want to use.
+3. They force an etcd redeployment which restarts the etcd pods which consume the new profile value.
+
+If the profile value is not valid, the configmap will be reset to the platform default before the etcd pods are started.
+
+#### Variation [optional]
+
+None
+
+### API Extensions
+
+None
+
+### Implementation Details/Notes/Constraints [optional]
+
+There will still be hardcoded values for each of the parameters and their mappings.
+We can reuse the functions in the Cluster Etcd Operator that retrieve the values, it would just be reading the environment variable for the profile, looking up the mapping, and returning the value.
+
+### Risks and Mitigations
+
+We will need to test the latency of the API server on platforms that currently use the default parameters to see if there are any issues with running slower values.
+If there is an issue, we could use slightly different values than proposed above; both Azure and IBMCloud VPC were originally meant to be temporary values to compensate for lacking IOPs.
```

Comment:
I would love to have some test utility here that allows us to inject host network latency and test the system against disruptions. I, for example, don't believe we need the "slow" profile for Azure anymore, but certainly for vsphere.

Would be good to have some utility to test the impact, and some success criterion (enough API availability?). Not sure this would be out of scope for the enhancement, but we definitely need that first before we should settle on any numbers. Even the current ones seem very arbitrary and dated to me.